### PR TITLE
[23] Implement search and filter of posted jobs page

### PR DESCRIPTION
**Pull request by priasdevo**
## What did you do

- image
- image
- Connect Filter Job API 
- Make responsive design of filter
- Refactor / Add props to old components to ready for filter

## Demo

- https://dev.modela.miello.dev/job

### CODE REVIEW

**Comment by miello** [comment]
image
Can we change it to 24hr format instead because it will more suitable when we look in the job post

File: apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts [hunk 1]
```diff
@@ -1,23 +1,72 @@
...
+import { IFilter, initialISearch, ISearch } from '../../types'
+
 const useJobListData = () => {
   const router = useRouter()
   const [job, setJob] = useState<GetJobCardWithMaxPageDto>()
   const [hasMore, setHasMore] = useState(true)
+  const [page, setPage] = useState(0)
   const pageControl = useRef(1)
+  const [search, setSearch] = useState<ISearch>(initialISearch)
+  const filterData = async (state: IFilter) => {
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/hooks/useJobListData/index.ts:16]: Consider wrap with `useCallback`

**Review by miello — CHANGES_REQUESTED** [review]
(Optional)
I think search filter can be considered as `form` so it might be able to reuse `FormController` to make it more compact

**Comment by priasdevo** [comment]
> (Optional) I think search filter can be considered as `form` so it might be able to reuse `FormController` to make it more compact

Though so as first but the search button and title searchBox is in the middle and the form is in the right side on desktop so i not sure if it can handle that ~~
In case u r w8, i will try to resolve comment at night td cause i am going to "เที่ยวว" ~~

File: apps/web/src/modules/job/list/pages/hooks/useMobileStyle/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useState } from 'react'
```
Comment (PKhing) [inline apps/web/src/modules/job/list/pages/hooks/useMobileStyle/index.ts:1]: consider using `useSwitch` (my bad, I forget to write it in change log)

Comment (priasdevo) [reply apps/web/src/modules/job/list/pages/hooks/useMobileStyle/index.ts:1]: oh, ok

File: apps/web/src/modules/job/list/pages/types.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  startDate: String | null
+  endDate: String | null
+  age: Number | null
+
+  minWage: Number | null
+  maxWage: Number | null
+
+  status: String[]
+  gender: String[]
+  castingId: Number | null
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/types.ts:33]:
```suggestion
  title: string | null
  startTime: string | null
  endTime: string | null
  location: string | null
  startDate: string | null
  endDate: string | null
  age: number | null

  minWage: number | null
  maxWage: number | null

  status: string[]
  gender: string[]
  castingId: number | null
```

File: apps/web/src/modules/job/list/pages/types.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+export interface IFilter {
+  title: String | null
+  startTime: Dayjs | null
+  endTime: Dayjs | null
+  location: String | null
+  startDate: Dayjs | null
+  endDate: Dayjs | null
+  wage: Number | null
+  deviant: Number | null
+  age: Number | null
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/types.ts:12]:
```suggestion
  title: string | null
  startTime: Dayjs | null
  endTime: Dayjs | null
  location: string | null
  startDate: Dayjs | null
  endDate: Dayjs | null
  wage: number | null
  deviant: number | null
  age: number | null
```

**Review by miello — CHANGES_REQUESTED** [review]
Last request changes for this PR

**Review by miello — APPROVED** [review]

**Pull request merged**